lua/tests/output-parser_spec.lua

```diff
@@ -1,239 +1,292 @@
-      '/home/cody/dev/src/git-down/src/test/kotlin/com/codymikol/state/neotest.kt::"a namespace"::"should handle passed assertions"'
-
-      it("should contain the correct result id", function()
-        assert.are.same(expected_id, result[expected_id].id)
-      end)
-    end)
-  end)
+	local output_parser = require("neotest-kotlin.src.output-parser")
+
+	describe("parse_lines", function()
+		it("complete example", function()
+			output_parser.parse_lines({
+				"> Task :app:cleanTest",
+				"> Task :app:checkKotlinGradlePluginConfigurationErrors",
+				"> Task :app:compileKotlin UP-TO-DATE",
+				"> Task :app:compileJava NO-SOURCE",
+				"> Task :app:processResources NO-SOURCE",
+				"> Task :app:classes UP-TO-DATE",
+				"> Task :app:compileTestKotlin UP-TO-DATE",
+				"> Task :app:compileTestJava NO-SOURCE",
+				"> Task :app:processTestResources NO-SOURCE",
+				"> Task :app:testClasses UP-TO-DATE",
+				"> Task :app:test",
+				"org.example.KotestDescribeSpec > a namespace > should handle failed assertions FAILED",
+				'   io.kotest.assertions.AssertionFailedError: expected:<"b"> but was:<"a">',
+				"       at app//org.example.KotestDescribeSpec$1$1$1.invokeSuspend(KotestDescribeExample.kt:9)",
+				"       at app//org.example.KotestDescribeSpec$1$1$1.invoke(KotestDescribeExample.kt)",
+				"org.example.KotestDescribeSpec > a namespace > should handle passed assertions PASSED",
+				"",
+				"org.example.KotestDescribeSpec > a namespace > should handle skipped assertions SKIPPED",
+				"",
+				"org.example.KotestDescribeSpec > a namespace > a nested namespace > org.example.KotestDescribeSpec.should handle failed assertions FAILED",
+				'   io.kotest.assertions.AssertionFailedError: expected:<"b"> but was:<"a">',
+				"       at app//org.example.KotestDescribeSpec$1$1$4$1.invokeSuspend(KotestDescribeExample.kt:22)",
+				"       at app//org.example.KotestDescribeSpec$1$1$4$1.invoke(KotestDescribeExample.kt)",
+				"       at app//org.example.KotestDescribeSpec$1$1$4$1.invoke(KotestDescribeExample.kt)",
+				"       at app//io.kotest.core.spec.style.scopes.DescribeSpecContainerScope$it$3.invokeSuspend(DescribeSpecContainerScope.kt:112)",
+				"org.example.KotestDescribeSpec > a namespace > a nested namespace > org.example.KotestDescribeSpec.should handle passed assertions PASSED",
+				"org.example.KotestDescribeSpec > a namespace > a nested namespace > org.example.KotestDescribeSpec.should handle skipped assertions SKIPPED",
+				"6 tests completed, 2 failed, 2 skipped",
+				"> Task :app:test FAILED",
+				"FAILURE: Build failed with an exception.",
+				"* What went wrong:",
+				"Execution failed for task ':app:test'.",
+				"> There were failing tests. See the report at: file:///home/nick/GitHub/neotest-kotlin/lua/tests/example_project/app/build/reports/tests/test/index.html",
+				"* Try:",
+				"> Run with --scan to get full insights.",
+				"BUILD FAILED in 4s",
+				"5 actionable tasks: 3 executed, 2 up-to-date",
```

Comment:
This is nearly all the gradle output, I trimmed down the stacktrace. Just an integration test on mock data, would make sense to do say run this on the `example_project` output as part of CI/CD in the future?